cmd/ketch/app_deploy.go

```diff
@@ -125,10 +137,53 @@ func appDeploy(ctx context.Context, cfg config, getImageConfigFile getImageConfi
 		deploymentSpec,
 	}
 	app.Spec.DeploymentsCount += 1
+
+	var watcher watch.Interface
+	if options.wait {
+		watcher, err = watchReconcileEvent(ctx, cfg.KubernetesClient(), &app)
+		if err != nil {
+			return err
+		}
+	}
+
 	if err = cfg.Client().Update(ctx, &app); err != nil {
 		return fmt.Errorf("failed to update the app: %v", err)
 	}
-	fmt.Fprintln(out, "Successfully deployed!")
+
+	// Await for reconcile result
+	if options.wait {
+		maxExecTime := time.NewTimer(time.Second * 20)
+		evtCh := watcher.ResultChan()
+		for {
+			select {
+			case evt, ok := <-evtCh:
+				if !ok {
+					err := errors.New("events channel unexpectedly closed")
+					return err
+				}
+				e, ok := evt.Object.(*corev1.Event)
+				if ok {
+					reason, err := controllers.ParseAppReconcileMessage(e.Reason)
+					if err != nil {
+						return err
+					}
+					if reason.DeploymentCount == app.Spec.DeploymentsCount {
+						switch e.Type {
+						case v1.EventTypeNormal:
+							fmt.Fprintln(out, "successfully deployed!")
+							return nil
+						case v1.EventTypeWarning:
+							return errors.New(e.Message)
+						}
+					}
+				}
+			case <-maxExecTime.C:
+				err := fmt.Errorf("maximum execution time exceeded")
+				return err
+			}
+		}
+	}
+	fmt.Fprintln(out, "successfully deployed!")
```

Comment:
is it true?

Reply:
If we are here - yes, this is the old part of the code, I just make it lowercase.